### PR TITLE
spacer 0.2.0 (new formula)

### PR DESCRIPTION
**Pull request by Moulick**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by Moulick** [comment]
Need some support with writing a good test though. Spacer is intended to be used while tailing logs and such. When tailing logs, it inserts some text to easily distinguish breaks in the log stream. Below is the easiest example on how the output looks like.

```shell
❯ for i in {0..4}; do sleep 1; echo $i ; done | spacer --after 1
0
2023-09-01 21:10:01 +01:00 2.0s ━━━━━━━━━━━━━━━
1
2023-09-01 21:10:02 +01:00 1.0s ━━━━━━━━━━━━━━━
2
2023-09-01 21:10:03 +01:00 1.0s ━━━━━━━━━━━━━━━
3
2023-09-01 21:10:05 +01:00 1.0s ━━━━━━━━━━━━━━━
4
```
What can I write as a test?

**Comment by Moulick** [comment]
@samwho Here we go

File: Formula/s/spacer.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    assert_match "spacer 0.2.0", shell_output("#{bin}/spacer --version").chomp
+
+    spacer_in = shell_output("for i in {0..3}; do sleep 1; echo $i ; done | #{bin}/spacer --after 0.5")
+    ohai spacer_in
+    assert_includes spacer_in, "+01:00"
```
Comment (p-linnane) [inline Formula/s/spacer.rb:19]:
```suggestion
    assert_match version.to_s, shell_output("#{bin}/spacer --version")

    output = shell_output("for i in {0..3}; do sleep 1; echo $i ; done | #{bin}/spacer --after 0.5")
    assert_match "1.0s", output
```

Comment (p-linnane) [reply Formula/s/spacer.rb:19]: After applying this suggestion please squash the commits into one with the message `spacer 0.2.0 (new formula)`.

Comment (Moulick) [reply Formula/s/spacer.rb]: I feel that "1.0s" is not reliable in the output, but the output always contains today's date. So I shifted the test to include that, is that ok? Also should I remove the `ohai` ?

**Comment by Moulick** [comment]
Linked: https://github.com/samwho/homebrew-spacer/pull/2

File: Formula/s/spacer.rb (hunk 1 above)
Comment (Moulick) [reply Formula/s/spacer.rb]: Also has to switch `{0..2}` to `((i=0; i<=2; i++))`; apparently does not work with whatever version of bash github actions uses, does work on my local bash though ¯\_(ツ)_/¯

Comment (p-linnane) [reply Formula/s/spacer.rb]: Just use whatever text is reliably printed. And yes, remove the `ohai` line, it's not necessary.

My suggested test worked on a couple runs using `zsh` on my system, but we should tweak if it isn't a reliable output.

Comment (Moulick) [reply Formula/s/spacer.rb]: It seems `homebrew/ubuntu22.04` runs the tests under `sh`, not even bash 🤔 . I'll remove `ohai` once I have it working in the CI

Comment (Moulick) [reply Formula/s/spacer.rb]: I caught a glimpse of how to test the linux builds in docker somwhere in the brew docs, can't find it again. Any idea where it is?

Comment (Moulick) [reply Formula/s/spacer.rb]: Ok now it's good to go :)

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@Moulick, thanks a lot for your contributions! 🎉

**Comment by Moulick** [comment]
Closes https://github.com/samwho/spacer/issues/11

**Pull request merged**